elasticapm/conf/__init__.py

```diff
@@ -10,8 +10,121 @@
 """
 
 import logging
+import os
+import re
+import socket
 
-__all__ = ('setup_logging', )
+from elasticapm.utils import compat
+
+__all__ = ('setup_logging', 'Config')
+
+
+class _ConfigValue(object):
+    def __init__(self, dict_key, env_key=None, type=compat.text_type, validators=None, default=None, doc=None):
+        self.type = type
+        self.dict_key = dict_key
+        self.validators = validators
+        self.default = default
+        self.doc = doc
+        if env_key is None:
+            env_key = 'ELASTIC_APM_' + dict_key
+        self.env_key = env_key
+
+    def __get__(self, instance, owner):
+        if instance:
+            return instance._values.get(self, self.default)
+        else:
+            return self.default
+
+    def __set__(self, instance, value):
+        if value is not None:
+            value = self.type(value)
+        instance._values[self] = value
+
+
+class _ListConfigValue(_ConfigValue):
+    def __init__(self, dict_key, list_separator=',', **kwargs):
+        self.list_serparator = list_separator
+        super(_ListConfigValue, self).__init__(dict_key, **kwargs)
+
+    def __set__(self, instance, value):
+        if isinstance(value, compat.string_types):
+            value = value.split(self.list_serparator)
+        else:
+            value = list(value) if value is not None else value
```

Comment:
why do you need to set `value` to `value` if it is `None`? How about: 
```
elif value not None: 
  value = list(value)
```

Reply:
that is indeed nicer, thanks!